### PR TITLE
Fix Kubernetes executor set wrong task status

### PR DESCRIPTION
**Pull request by pankajastro**
In the case of multiple schedulers lots of tasks running
If schedulers restart a bit quickly some cases it sets the wrong task status.
In this PR, I'm changing some checks so that if the pod status is non-terminal 
then set the task status to Failed only if the deletion_timestamp (metadata.deletion_timestamp) is set for the pod. This gets set when server receive the deletion request https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-meta/#ObjectMeta

**Example screenshot before changes** 
Screenshot 2023-05-14 at 2 55 02 AM

**Example screenshot after changes** 
Screenshot 2023-05-14 at 2 49 30 AM

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/executors/kubernetes_executor.py [hunk 1]
```diff
@@ -217,8 +217,11 @@ def process_status(
         event: Any,
     ) -> None:
         """Process status response."""
+        pod = event["object"]
         if status == "Pending":
-            if event["type"] == "DELETED":
+            # deletion_timestamp is set by kube server when a graceful deletion is requested.
+            # since the pod is in intermediate state so set TI status failed
+            if pod.metadata.deletion_timestamp is not None:
```
Comment (ephraimbuddy) [inline airflow/executors/kubernetes_executor.py]:
```suggestion
            if event["type"] == "DELETED" and pod.metadata.deletion_timestamp is not None:
```
WDYT?

File: airflow/executors/kubernetes_executor.py [hunk 2]
```diff
@@ -246,7 +249,9 @@ def process_status(
             self.log.info("Event: %s Succeeded", pod_name)
             self.watcher_queue.put((pod_name, namespace, None, annotations, resource_version))
         elif status == "Running":
-            if event["type"] == "DELETED":
+            # deletion_timestamp is set by kube server when a graceful deletion is requested.
+            # since the pod is in intermediate state so set TI status failed
+            if pod.metadata.deletion_timestamp is not None:
```
Comment (ephraimbuddy) [inline airflow/executors/kubernetes_executor.py]:
```suggestion
            if event["type"] == "DELETED" and pod.metadata.deletion_timestamp is not None:
```
Not sure if the `is not None` is necessary

**Review by ephraimbuddy — APPROVED** [review]
LGTM. You can test this on a deployment. Please revert when tested as I would like this to be part of 2.6.2 which we will cut tomorrow

**Comment by dirrao** [comment]
#32249 issue fixed in this MR.

**Pull request merged**